hooks: update QtNetwork hooks for Qt 6.2 compatibility

rokm opened the pull request:
Qt 6.2 introduced new set of `tls` plugins that need to be collected for `QSslSocket` to work.

rokm committed:
hookutils: qt: fix filtering of debug plugin variants

Simply filtering out "*d.dll" is insufficient as
we now have Qt plugins whose names end with
"backend" (e.g., "tls/qopensslbackend.dll").

Therefore, upgrade the filtering logic to exclude
a DLL whose name ends with "d" only if the
corresponding "release" name (i.e., name shortened
by one character) also exists.

rokm committed:
hookutils: qt: collect new QtNetwork plugins

As of Qt 6.2, we need to collect "tls" plugins as part of QtNetwork
for QSslSocket to work.

There is another group of plugins that belongs to QtNetwork in Qt 6.2,
called "networkinformation". These were introduced in Qt 6.1 as
"networkinformationbackends". So collect both variants.

rokm committed:
tests: repin pyside6 and pyqt6 to 6.2.0

rokm committed:
ci: do not use QT_DEBUG_PLUGINS=1

We currently set QT_DEBUG_PLUGINS=1, but only on ubuntu runner.
We should either use it on all OSes, or just enable the extra
verbosity when we suspect a problem with plugin loading instead
of having it always on.

rokm committed:
ci: set QT_QPA_PLATFORM=offscreen

Run Qt-based tests with offscreen backend instead of default
display one. Should both speed up the tests that display the UI,
as well as make them more resilient to CI pipeline's display issues
(e.g., lack of OpenGL on macOS runners).

rokm commented:
The switch to offscreen Qt backend will hopefully take care of failures such as seen [here](https://github.com/pyinstaller/pyinstaller/actions/runs/1312455374) and [here](https://github.com/pyinstaller/pyinstaller/actions/runs/1312455374/attempts/1) (two subsequent attempts). I think for our testing purposes, using offscreen backend makes more sense than hoping that display works properly on the CI. This approach is also suggested in #5120.

bwoodsend reviewed (approved):
Looks good to me.

bwoodsend commented on tests/requirements-libraries.txt:
```diff
@@ -31,12 +31,10 @@ future==0.18.2
 pyqtwebengine==5.15.4
-# NOTE: keep pyqt6-qt6 pinned together with pyqt6 until we switch to 6.2.0
-pyqt6==6.1.0  # pyup: ignore
-pyqt6-qt6==6.1.0  # pyup: ignore
```
Is there a reason you unpinned `pyqt6-qt6` instead of repinning it to 6.2?

rokm replied:
Mainly because it works now without it again. But you're right, that's only as long as the versions remain in sync...

bwoodsend commented:
Do you want to try just setting the timeout limit higher for that dodgy QtQml test? The variants that are passing are all being flagged up as very slow tests getting close to the 300 seconds mark.

rokm commented:
> Do you want to try just setting the timeout limit higher for that dodgy QtQml test? The variants that are passing are all being flagged up as very slow tests getting close to the 300 seconds mark.

Ah, that's a good catch, thanks! 

I had re-run it again to see if it will consistently fail on that particular combination again, so I can try reproducing it locally with the same python version. 

But if it's a timing issue (slow build and/or slow startup), then we should just raise the timeout limit.

bwoodsend commented:
Also [this](https://github.com/pyinstaller/pyinstaller/runs/3836620743?check_suite_focus=true#step:12:1805) makes me think that that Windows + Python 3.7 + pandas sporadic failure we've been chasing for ages is also *just a bit slow*.

rokm committed:
tests: double the global timeout to 10 minutes

For some tests, 5 minutes seems to be cutting it too close... so
increase the safety margin.
